sklearn/naive_bayes.py

```diff
@@ -522,7 +523,7 @@ def fit(self, X, y, sample_weight=None):
 
         # convert to float to support sample weight consistently;
         # this means we also don't have to cast X to floating point
-        Y = Y.astype(np.float64)
+        Y = astype(Y, np.float64, copy=False)
```

Comment:
Same remark as https://github.com/scikit-learn/scikit-learn/pull/4645/files#r29588000 but for `LabelBinarizer()` instead of `label_binarize`.